Library/Homebrew/dev-cmd/extract.rb

```diff
@@ -97,47 +97,64 @@ def extract
     # Expect exactly two named arguments: formula and tap
     raise UsageError if ARGV.named.length != 2
 
+    if ARGV.named.first !~ HOMEBREW_TAP_FORMULA_REGEX
+      name = ARGV.named.first.downcase
+      source_tap = CoreTap.instance
+    else
+      name = Regexp.last_match(3).downcase
+      source_tap = Tap.fetch(Regexp.last_match(1), Regexp.last_match(2))
+      raise TapFormulaUnavailableError.new(source_tap, name) unless source_tap.installed?
+    end
+
     destination_tap = Tap.fetch(ARGV.named.second)
     odie "Cannot extract formula to homebrew/core!" if destination_tap.core_tap?
+    odie "Cannot extract formula to the same tap!" if destination_tap == source_tap
     destination_tap.install unless destination_tap.installed?
 
-    name = ARGV.named.first.downcase
-    repo = CoreTap.instance.path
-    # Formulae can technically live in "<repo>/<formula>.rb" or
-    # "<repo>/Formula/<formula>.rb", but explicitly use the latter for now
-    # since that is how the core tap is structured.
-    file = repo/"Formula/#{name}.rb"
+    repo = source_tap.path
+    pattern = source_tap.core_tap? ? repo/"Formula/#{name}.rb" : repo/"{,**/}#{name}.rb"
 
     if args.version
       ohai "Searching repository history"
       version = args.version
       rev = "HEAD"
       test_formula = nil
+      result = ""
       loop do
-        loop do
-          rev = Git.last_revision_commit_of_file(repo, file, before_commit: "#{rev}~1")
-          break if rev.empty?
-          break unless Git.last_revision_of_file(repo, file, before_commit: rev).empty?
+        rev, (path,) = Git.last_revision_commit_of_files(repo, pattern, before_commit: "#{rev}~1")
```

Comment:
Why the `()`?

Reply:
`last_revision_commit_of_files` return a list of files, i.e. `[rev, [file1, file2,...]]`. Here just need to get the 1 file.

Reply:
Cool. Never seen that syntax before.

Reply:
I knew this from python. Just tried it in ruby and it works :)

Reply:
Heh nice 👏